agent/acs/handler/acs_handler.go

```diff
@@ -72,13 +71,43 @@ type StartSessionArguments struct {
 	AcceptInvalidCert    bool
 }
 
+// sessionResources defines the resource creator interface for starting
+// a session with ACS. This interface is intended to define methods
+// that create resources used to establish the connection to ACS
+// It is confined to just the createACSClient() method for now. It can be
+// extended to include the acsWsURL() and newDisconnectionTimer() methods
+// when needed
+// The goal is to make it easier to test and inject dependencies
+type sessionResources interface {
+	// createACSClient creates a new websocket client
+	createACSClient(url string) wsclient.ClientServer
+}
+
+// acsSessionResources implements the resource creator interface to
+// create resources needed to connect to ACS
+type acsSessionResources struct {
+	startSessionArguments StartSessionArguments
+}
+
 // StartSession creates a session with ACS and handles requests from ACS.
-// It also tries to repeatedly connect to ACS when disconnected.
+// It creates resources required to invoke the package scoped 'startSession()'
+// method and invokes the same to repeatedly connect to ACS when disconnected
 func StartSession(ctx context.Context, args StartSessionArguments) error {
 	backoff := utils.NewSimpleBackoff(connectionBackoffMin, connectionBackoffMax, connectionBackoffJitter, connectionBackoffMultiplier)
+	session := newACSSessionResources(args)
+	return startSession(ctx, args, backoff, session)
+}
 
+// newACSSessionResources creates a new acsSessionResources object
+func newACSSessionResources(args StartSessionArguments) sessionResources {
```

Comment:
Can't think of a compelling argument to keep it around. Might make sense when we have more things to abstract out. I'll modify this in a bit.